core/server/src/main/java/alluxio/master/AbstractMaster.java

```diff
@@ -285,4 +285,35 @@ protected void waitForJournalFlush(long counter) {
   protected ExecutorService getExecutorService() {
     return mExecutorService;
   }
+
+  /**
+   * @return new instance of {@link JournalContext}
+   */
+  protected JournalContext createJournalContext() {
+    return new JournalContext();
+  }
+
+  /**
+   * Context for storing journaling information.
+   */
+  public final class JournalContext implements AutoCloseable {
+    private long mFlushCounter;
+
+    private JournalContext() {
+      mFlushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
```

Comment:
NIT: I guess we could make `AsyncJournalWriter.INVALID_FLUSH_COUNTER` private now since it becomes completely an internal state. Up to you